src/DependencyInjection/Factory/RoutingAdminFactory.php

```diff
@@ -35,16 +35,22 @@ public function getKey()
     public function addConfiguration(NodeBuilder $builder)
     {
         $builder->scalarNode('basepath')->defaultNull()->end();
+        $builder->scalarNode('content_basepath')->defaultNull()->end();
     }
 
     /**
      * {@inheritdoc}
      */
     public function create(array $config, ContainerBuilder $container, XmlFileLoader $loader)
     {
+        if (!$config['enabled']) {
```

Comment:
> (we check if one of the parameters that would be set if enabled is indeed set)

That's normal behaviour for compiler passes. Imo, we should keep it. Sometimes, the factory is enabled but a specific feature isn't, so the compiler pass shouldn't be executed, etc. Let's keep all execution logic in the pass itself.